pkg/util/strategicpatch/patch.go

```diff
@@ -90,6 +121,7 @@ func mergeMap(original, patch map[string]interface{}, t reflect.Type) (map[strin
 		// original. Otherwise, skip it.
 		if patchV == nil {
 			if _, ok := original[k]; ok {
+				dw.WriteRemoval(fmt.Sprintf("\"%s\": %v", k, original[k]))
 				delete(original, k)
```

Comment:
@kbeecher, it looks like this line causes problem if we are generating diff between `original` and `patch`, suppose the `original` is a map:
a: 1
b: 2
c: 3, 
and the `patch` is also a map:
a: nil,
then this PR will generate
-a: 1,
but that's not the `diff` between `original` and `patch`.

Actually I don't know what the `diff` should be in this case. Should `original`+`diff` = `patch` under strategicmerge's syntax? If so, there seems no proper `diff` can result in "a: nil". I'm really confused now :(